pkg/k8s/errors.go

```diff
@@ -233,3 +233,18 @@ func (e *ImmutableFieldsMutationError) Error() string {
 func NewImmutableFieldsMutationError(immutableFields []string) *ImmutableFieldsMutationError {
 	return &ImmutableFieldsMutationError{immutableFields}
 }
+
+type UpdateNotSupportedError struct {
+	resourceGVK schema.GroupVersionKind
+	namespace   types.NamespacedName
+	resource    string
```

Comment:
In the implementation, this field contains the relative resource name of the GCP resource. I'm hesitant to mention it in the error message because this error is at the KCC level, not the GCP level. If one day the API team starts supporting update for the resource, the error message might become confusing because KCC is saying that an updatable GCP resource cannot be updated.

Reply:
Done